### ISSUE TITLE
Documentation⁺

### ISSUE THREAD

**Issue by lindsaycarr** [USGS-R/training-curriculum#156]
1. Distinguish the different types of documentation for R packages.
2. Develop documentation for individual functions.
3. Create a vignette to highlight the top-level package uses.
4. Edit and update README files.

`date: "9999-07-31"`

**Comment by lindsaycarr** [comment]
- roxygen syntax, use markdown syntax rather than LaTex? Limits examples we can show from our own packages. 
- **EXAMPLES** in your docs
- everything in testthat are in examples

**Comment by ldecicco-USGS** [comment]
As a maintainer for dataRetrieval....I think it's very important that we add info to the "dontrun" section. Currently it says:
```
Sometimes people use `\dontrun{}` to show examples of code that will cause in error. To use this, add your example code that will cause an error inside of the `{ }`.
```
There are a bunch of reasons to put code in the dontrun bracket, and I have never heard of people using it to show how to create an error. Long running code, code that requires user-inputs, code that depends on a web service call those have always been the reasons I've heard of people using dontrun.

Basically, I just want to expand the ambassadors that are aware that ALL of the dataRetrieval examples are surrounded by `dontrun`...but totally should be considered valid examples, and this is because of CRAN requirements.

**Comment by ldecicco-USGS** [comment]
Something I'm learning through painful experience....we can't use `#` in a url in the helpfiles. It's also annoying that "Check" isn't catching those errors on 3.4.0, but at least on `rhub`, `check_with_rdevel()` does clue us in. Interestingly though I guess, CRAN's regular 3.4.0 is also flagging the warnings, so it's not just `rdevel`.

Anyway, probably worth mentioning in here.

**Comment by lindsaycarr** [comment]
Still need to add something about how to build vignettes:

> ldecicco: So here's the thing. If you have a .Rmd file in the vignette's folder, and you only count on roxygen...it will create the .R and .html files, but you still have to then manually move them from the vignette folder to the inst/doc folder. If you just use devtools::build_vignette(), it does both of those steps. I guess there's also a way to use a make file in your vignette folder, but I'd still vote the devtools function is the easiest.

**Comment by lindsaycarr** [comment]
Content complete, further feedback provided in #206
